src/app/stream-deck/actions.ts

```diff
@@ -0,0 +1,366 @@
+  'loadout' | 'item'
+>();
+
+export const streamDeckClearSelection = createAction('stream-deck/CLEAR-SELECTION')();
+
+// serialize the data and send it if connected
+export function sendToStreamDeck(args: Record<string, any>): ThunkResult {
+  return async () => {
+    if (streamDeckWebSocket?.readyState === WebSocket.OPEN) {
+      const sharedKey = streamDeckLocal.sharedKey();
+      sharedKey && streamDeckWebSocket.send(encryption.encrypt(JSON.stringify(args), sharedKey));
+    }
+  };
+}
+
+// on click on InventoryItem send the selected item to the Stream Deck
+export function streamDeckSelectItem(item: DimItem): ThunkResult {
+  return async (dispatch, getState) => {
+    const { streamDeck } = getState();
+    if (streamDeck.enabled && streamDeck.selection === 'item') {
+      hideItemPopup();
+      streamDeck.selectionPromise.resolve();
+      dispatch(streamDeckClearSelection());
+      return dispatch(
+        sendToStreamDeck({
+          selectionType: 'item',
+          selection: {
+            label: item.name,
+            subtitle: item.typeName,
+            item: item.id,
+            icon: item.icon,
+          },
+        })
+      );
+    }
+  };
+}
+
+function findSubClass(items: LoadoutItem[], state: RootState) {
+  const defs = d2ManifestSelector(state);
+  for (const item of items) {
+    const def = defs?.InventoryItem.get(item.hash);
+    // find subclass item
+    if (def?.inventory?.bucketTypeHash === BucketHashes.Subclass) {
+      return def.displayProperties.icon;
+    }
+  }
+}
+
+// on click on LoadoutView send the selected loadout and the related character identifier to the Stream Deck
+export function streamDeckSelectLoadout(loadout: Loadout, store: DimStore): ThunkResult {
+  return async (dispatch, getState) => {
+    const state = getState();
+    if (state.streamDeck.enabled && state.streamDeck.selection === 'loadout') {
+      state.streamDeck.selectionPromise.resolve();
+      dispatch(streamDeckClearSelection());
+      return dispatch(
+        sendToStreamDeck({
+          selectionType: 'loadout',
+          selection: {
+            label: loadout.name,
+            loadout: loadout.id,
+            subtitle: store.className ?? loadout.notes,
+            character: store.id,
+            icon: findSubClass(loadout.items, state),
+          },
+        })
+      );
+    }
+  };
+}
+
+// Show notification asking for selection
+function showSelectionNotification(
+  state: RootState,
+  selectionType: 'item' | 'loadout',
+  onCancel?: () => void
+) {
+  showNotification({
+    title: 'Elgato Stream Deck',
+    body:
+      selectionType === 'item' ? t('StreamDeck.Selection.Item') : t('StreamDeck.Selection.Loadout'),
+    type: 'info',
+    duration: 500,
+    onCancel,
+    promise: state.streamDeck.selectionPromise.promise,
+  });
+}
+
+// click on the nav link to open a specific page
+function goToPage(path: string) {
```

Comment:
You can directly use the router for this rather than clicking links

Reply:
That's a reasonable way to go. I know there are some plugins that let you bind router state to redux state that help with this as well. We just hadn't needed it before.